### PR TITLE
Remove TDD docs, add more info on running tests

### LINKED ISSUE #2295: Update dev documentation on how to run 1 test
[kind/Documentation]

## What did you find missing in the documentation?

We should add commands such as this:

`ginkgo -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=1 -progress -v tests/integration/ -F "FOOBAR"` to dev docs

and modifying integration tests to use "FIt" instead of "It" if you choose to run only 1 test

## What is the relevance of it?

Being able to run integration tests locally without running the whole suite.

### PR DESCRIPTION
**Pull request by cdrage**
**What kind of PR is this?**

/kind docs

**What does does this PR do / why we need it**:

- Removes TDD documentation which is outdated / we no longer use it.
- Adds documentation on how to run a single test

**Which issue(s) this PR fixes**:

Fixes #2295

**How to test changes / Special notes to the reviewer**:

N/A

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

File: docs/dev/development.adoc [hunk 1]
```diff
@@ -181,7 +182,16 @@ $ MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags "--ena
...
+Integration tests utilize link:https://github.com/onsi/ginkgo[`Ginkgo`] and it's preferred matcher library link:https://github.com/onsi/gomega[`Gomega`] which define sets of test cases (spec). As per ginkgo test file comprises specs and these test file are controlled by test suite. 
+
+Test and test suite files are located in `tests/integration` directory and can be called using `make test-integration`. 
+
+To run ONE individual test, you can either:
+
+* Supply the name via command-line: `ginkgo -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=1 -progress -v tests/integration/ -F "When executing catalog list without component directory"`
+* Modify the `It` statement to `Fit` and run `ginkgo -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=1 -progress -v tests/integration/`
+
+Integration tests validates and focuses on specific fields of odo functionality or individual commands. For example, `cmd_app_test.go` or `generic_test.go`.
```
Comment (girishramnani) [inline docs/dev/development.adoc]: validate and focus

File: docs/dev/development.adoc [hunk 2]
```diff
@@ -181,7 +182,16 @@ $ MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags "--ena
...
+Integration tests utilize link:https://github.com/onsi/ginkgo[`Ginkgo`] and it's preferred matcher library link:https://github.com/onsi/gomega[`Gomega`] which define sets of test cases (spec). As per ginkgo test file comprises specs and these test file are controlled by test suite. 
+
+Test and test suite files are located in `tests/integration` directory and can be called using `make test-integration`. 
+
+To run ONE individual test, you can either:
+
+* Supply the name via command-line: `ginkgo -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=1 -progress -v tests/integration/ -F "When executing catalog list without component directory"`
```
Comment (amitkrout) [inline docs/dev/development.adoc]: Command does not work
```
$ ginkgo -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=1 -progress -v tests/integration/ -F "When executing catalog list without component directory"
Running Suite: Integration Suite
================================
Random Seed: 1572505443 - Will randomize all specs
Will run 122 of 122 specs         <------------------------------------ It running all the specs

odo push command tests when push command is executed 
  should not build when changes are detected in a ignored file
[...]
```

Comment (amitkrout) [reply docs/dev/development.adoc]: In this case let make it simple
```
* Supply the name via command-line: `ginkgo -focus="<spec title>" test_dir_path_of_spec_file`
+

----
ginkgo -focus="When executing catalog list without component directory" tests/integration/
----
+
```

In a NOTE mention that ```To customize output use ginkgo flags. see `ginkgo -h` for more details```

Comment (amitkrout) [reply docs/dev/development.adoc]: Verified the command locally
```
$ ginkgo -focus="When executing catalog list without component directory" tests/integration/
Running Suite: Integration Suite
================================
Random Seed: 1572506506
Will run 1 of 122 specs      <---------------------- Running only one spec 
[...]
```

File: docs/dev/development.adoc [hunk 3]
```diff
@@ -181,7 +182,16 @@ $ MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags "--ena
...
+* Modify the `It` statement to `Fit` and run `ginkgo -randomizeAllSpecs -slowSpecThreshold=120 -timeout 7200s -nodes=1 -progress -v tests/integration/`
```
Comment (amitkrout) [inline docs/dev/development.adoc]: You can make it simple and generic like
```
* Modify the `It` statement to `Fit` and run `ginkgo test_dir_path_of_spec_file`
+
----
For example:
ginkgo tests/integration/
----
+

As i mentioned above you can uses a common NOTE of ``` NOTE: To customize output use ginkgo flags. see `ginkgo -h` for more details``` for both the steps

Comment (amitkrout) [reply docs/dev/development.adoc]: Verified locally
```
$ ginkgo tests/integration/
Running Suite: Integration Suite
================================
Random Seed: 1572507116
Will run 1 of 122 specs    --------------- Running only one spec
[...]
```

**Comment by amitkrout** [comment]
#2330 will fix the flake observed in ci/prow/v4.1-integration

**Comment by cdrage** [comment]
ping @dharmit @mohammedzee1000 @kadel

File: docs/dev/development.adoc [hunk 4]
```diff
@@ -80,19 +80,20 @@ $ git remote set-url --push upstream no_push
...
-* Avoid merging the PR manually, unless it is an emergency and  you have the required permissions). Prow’s tide component automatically merges the PR once all the conditions are met.
+* Avoid merging the PR manually (unless it is an emergency and  you have the required permissions). Prow’s tide component automatically merges the PR once all the conditions are met.
 It also ensures that post-submit tests (tests that run before merge) validate the PR.
 * Use the link:https://deck-ci.svc.ci.openshift.org/command-help[command-help] to see the list of possible bot commands.
 
-== Test Driven Development
+== Tests
 
-We follow the Test Driven Development(TDD) workflow in our development process. You can read more about it link:tdd-workflow.adoc[here].
+We use both unit, integration and e2e (End-2-End) tests.
```
Comment (dharmit) [inline docs/dev/development.adoc]: Aren't those three kind of tests? Referring to the word `both` being used in that statement.

File: docs/dev/development.adoc [hunk 5]
```diff
@@ -181,7 +182,16 @@ $ MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags "--ena
 
 *Integration tests:*
 
-Integration tests utilize link:https://github.com/onsi/ginkgo[`Ginkgo`] and it's preferred matcher library link:https://github.com/onsi/gomega[`Gomega`] which define sets of test cases (spec). As per ginkgo test file comprises specs and these test file are controlled by test suite. Test and test suite files are located in `tests/integration` directory and can be called using `.PHONY` within `makefile`. Integration tests validates and focuses on specific fields of odo functionality or individual commands. For example, `cmd_app_test.go` or `generic_test.go`.
+Integration tests utilize link:https://github.com/onsi/ginkgo[`Ginkgo`] and it's preferred matcher library link:https://github.com/onsi/gomega[`Gomega`] which define sets of test cases (spec). As per ginkgo test file comprises specs and these test file are controlled by test suite.
```
Comment (dharmit) [inline docs/dev/development.adoc]: `s/it's preferred/its preferred`?

**Comment by cdrage** [comment]
Update, thanks @dharmit !

**Pull request merged**